core/src/main/java/org/mule/runtime/core/internal/exception/ErrorHandler.java

```diff
@@ -181,4 +185,13 @@ public void setStatistics(FlowConstructStatistics flowStatistics) {
     }
   }
 
+  public ErrorHandler copy(Location flowLocation) {
```

Comment:
`localizeFor` instead?
Add a javadoc explaining that this is only to be called for global error handlers (and add the validation)